### PR TITLE
Added first version of source import

### PR DESCRIPTION
**Pull request by sebastianberm**
This might not be 100% complete yet, but it should allow for source imports to take place.

### CODE REVIEW

**Comment by acrobat** [comment]
Closing and reopening this PR to trigger the styleci build.

**Comment by sebastianberm** [comment]
I think there is something wrong with the styleci check. (This isn't the only PR where this seems to be _hanging_).

**Comment by GrahamCampbell** [comment]
The StyleCI check is not hanging. It has not been triggered at all. I just tried to re-issue StyleCI's webhook, but access was denied. There are two possible reasons for this:

1. Access to **KnpLabs** could have been revoked in the third party access control settings.
2. The repo is linked to **cursedcoder** who might not be an admin on the repo.

**Comment by acrobat** [comment]
The styleci integration is working again, there was an issue with linked user not having the required permission on github. (thanks @GrahamCampbell for the quick answer!)

@sebastianberm Can you check the issues styleci has found? After those are fixed I will review this PR, thanks!

File: lib/Github/Api/Migrations/SourceImport.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    use AcceptHeaderTrait;
+
+    /**
+     * Start import of repo from other VCS.
+     *
+     * @link https://developer.github.com/v3/migrations/source_imports/
+     *
+     * return array
+     */
+    public function start($params, $owner, $repoName)
```
Comment (acrobat) [inline lib/Github/Api/Migrations/SourceImport.php]: I would move the `$params` parameter to the end of the function parameter list, to be in line with the rest of the code. This parameter can also be typehinted as array!

File: lib/Github/Api/Migrations/SourceImport.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+use Github\Api\AcceptHeaderTrait;
+use Github\Exception\InvalidArgumentException;
+use Github\Exception\MissingArgumentException;
+
+/**
+ * Support for Migrations.
+ *
+ * @link   http://developer.github.com/v3/migrations
+ *
+ * @author Sebastian Berm <sebastian@sebsoft.nl>
```
Comment (acrobat) [inline lib/Github/Api/Migrations/SourceImport.php]: Also add the `@experimental` annotation, to indicate this is a preview endpoint. See #806

**Comment by acrobat** [comment]
@sebastianberm Can you also add some basic tests for the new endpoint? You can see [an example here](https://github.com/KnpLabs/php-github-api/blob/4a2a77b1f68f85520fcc948d7a59b5aa55d7b7f2/test/Github/Tests/Api/RepoTest.php).

**Comment by GrahamCampbell** [comment]
This needs re-targetting to 2.x.

**Comment by acrobat** [comment]
Development has stalled in this PR, so I'm closing this for now. @sebastianberm if you still have time to finish it, please rebase against the current master, apply the requested changes and reopen the PR or if anyone else needs this feature please open a new PR. Thanks!

**Pull request closed without merging**